mongorestore/mongorestore_test.go

```diff
@@ -211,3 +211,69 @@ func TestMongorestorePreserveUUID(t *testing.T) {
 
 	})
 }
+
+// test --maintainInsertionOrder and --stopOnError behavior
+func TestMongorestoreMIOSOE(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	client, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+	database := client.Database("miodb")
+	coll := database.Collection("mio")
+
+	Convey("default restore ignores dup key errors", t, func() {
+		restore, err := getRestoreWithArgs("testdata/10k1dup10k.bson",
+			CollectionOption, coll.Name(),
+			DBOption, database.Name(),
+			DropOption)
+		So(err, ShouldBeNil)
+		So(restore.OutputOptions.MaintainInsertionOrder, ShouldBeFalse)
+
+		err = restore.Restore()
```

Comment:
Similar import testing gets a success/failed count for test assertions, not just relying on count.  Should these tests do the same?